property_caching/__init__.py

```diff
@@ -11,10 +11,17 @@ def _class_cached_property(self):
     return property(update_wrapper(_class_cached_property, fn))
 
 
-def cached_property(fn):
-    def _cached_property(self):
-        return _get_property_value(fn, self, _OBJ_CACHE_ATTR_NAME)
-    return property(update_wrapper(_cached_property, fn))
+def cached_property(fn=None, cache_none=True):
+    def cached_property_decorator(fn):
+        def _cached_property(self):
+            return _get_property_value(
+                fn, self, _OBJ_CACHE_ATTR_NAME, cache_none)
+        return property(update_wrapper(_cached_property, fn))
+
+    # When used without arguments (@cached_property), behave like a normal
+    # decorator: take a fn and return a wrapper.  When used with arguments
+    # (@cached_property(cache_none=True)), return a decorator to take the fn.
+    return cached_property_decorator(fn) if fn else cached_property_decorator
```

Comment:
Yeah, new decorator would be better. The big fat comment should have alerted me this was being too clever.

Now that I think about it, what about `truthy_cached_property`? It would only cache truthy values?